src/pages/ProfilesExplorerView/components/SceneProfilesExplorer/SceneProfilesExplorer.tsx

```diff
@@ -225,6 +223,7 @@ export class SceneProfilesExplorer extends SceneObjectBase<SceneProfilesExplorer
     item?: GridItemData;
   }) {
     if (comesFromUserAction) {
+      prepareHistoryEntry();
```

Comment:
could this not just do a locationService.partial as well (you would get history step and update state in one action)

Reply:
this is what we do for most URL synced "view" state in core dashboards (update the URL via normal link or locatinService)

Reply:
You mean to update the URL instead of changing the state and let the `updateFromUrl` handle the update? 🤔  Does it mean you need to know the final name of the variable when doing `locationService.partial()`? How do you handle it if variable name may be different after processing with `urlKeyMapper.getUniqueKey` in UrlSyncManager?

Reply:
true, you would need t know the real URL key (so would only work for things where you know this). 

Was just an idea to automatically get a history step for a user action 

been making progress on https://github.com/grafana/scenes/pull/760 , can you check it out?